test/component/CustomsDeclarationStatusSpec.scala

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package component
+
+import com.github.tomakehurst.wiremock.client.WireMock.{postRequestedFor, urlEqualTo, verify}
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach, Matchers, OptionValues}
+import play.api.mvc._
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{status, _}
+import uk.gov.hmrc.customs.declaration.model.{ApiSubscriptionKey, VersionOne, VersionThree, VersionTwo}
+import util.FakeRequests._
+import util.RequestHeaders.{ValidHeadersV2, ValidHeadersV3}
+import util.externalservices.{ApiSubscriptionFieldsService, AuthService, GoogleAnalyticsService, MdgStatusDeclarationService}
+import util.{AuditService, CustomsDeclarationsExternalServicesConfig}
+
+import scala.concurrent.Future
+
+class CustomsDeclarationStatusSpec extends ComponentTestSpec with AuditService with ExpectedTestResponses
+  with Matchers
+  with OptionValues
+  with BeforeAndAfterAll
+  with BeforeAndAfterEach
+  with MdgStatusDeclarationService
+  with ApiSubscriptionFieldsService
+  with AuthService
+  with GoogleAnalyticsService {
+
+  private val mrn = "some-mrn"
+  private val endpoint = s"/status-request/mrn/$mrn"
+
+  private val apiSubscriptionKeyForXClientIdV1 =
+    ApiSubscriptionKey(clientId = clientId, context = "customs%2Fdeclarations", version = VersionOne)
+
+  private val apiSubscriptionKeyForXClientIdV2 = apiSubscriptionKeyForXClientIdV1.copy(version = VersionTwo)
+
+  private val apiSubscriptionKeyForXClientIdV3 = apiSubscriptionKeyForXClientIdV2.copy(version = VersionThree)
+
+  private val validResponse =
+    """<stat:declarationManagementInformationResponse xmlns:stat="http://gov.uk/customs/declarations/status-request">
```

Comment:
I would keep this as an XML node if possible rather than string

Reply:
i'm testing the exact layout of the xml as it is printed to the response body so i'd rather leave it as a string